src/lexer/token.py

```diff
@@ -1,43 +1,122 @@
+from constants.constants import DELIMS
+from enum import Enum
+
+
+class TokenType(Enum):
+    def __init__(self, token: str, delim_id: str):
+        self._token = token
+        self.delim_id = delim_id
+
+        self.expected_delims = DELIMS[self.delim_id]
+
+    @property
+    def token(self):
+        return self._token
+
+    def __str__(self):
+        return self.token
```

Comment:
Noted dun sa properties.

Dun sa `__str__` dunder, I added that because in our test run ng lexer, ganto siya mag-print ng tokens.
```
    for token in x.tokens:
        print(f"{token.lexeme}", end="\t")
        print(f"{token.token}", end="\t")
        print(f"{token.position} - {token.end_position}", end="\t")
        print()
```

The `print(f"{token.token}", end="\t")` part is what prints the token type. If want niyo alisin si `__str__`, we will access the token type with `token.token.token`, which seems awkward to me HAHAHAHAHA